### PR TITLE
Update spatiotemporal chapter

### PR DESCRIPTION
**Pull request by pat-s**
To reflect {mlr3spatiotempcv} 2.0 changes.

@be-marc Review appreciated!

We also need to add more content about {mlr3spatial}. I've also now split the content into

- perf. estimation with nested CV ({mlr3spatiotempcv})
- spatial prediction ({mlr3spatial})

{mlr3spatiotempcv} still needed as there were some issues with the return of the cstf autoplot() fun.

### CODE REVIEW

**Comment by pat-s** [comment]
@be-marc I'd like to merge this very soon to have a working documentation for the current {mlr3spatiotempcv} release. Review still welcome, we can do a follow-up PR if needed :)

File: book/08-special-spatiotemp.qmd [hunk 1]
```diff
@@ -132,70 +167,209 @@ rr_sp = resample(
...
 
-Note that setting the correct CRS for the given data is important which is done during task creation
+:::{.callout-warning}
+Note that setting the correct CRS for the given data is important which is done **during task creation**.
 Spatial offsets of up to multiple meters may occur if the wrong CRS is supplied initially.
+:::
 
-This example used an already created task via the sugar function `r ref("tsk()")`.
+This example used a built-in mlr3 task via `r ref("tsk()")`.
 In practice however, one needs to create a spatiotemporal task via `r ref("TaskClassifST")`/`r ref("TaskRegrST")` and set the `crs` argument.
```
Comment (be-marc) [inline book/08-special-spatiotemp.qmd:208]: or is extracted from the sf object.

File: book/08-special-spatiotemp.qmd [hunk 2]
```diff
@@ -132,70 +167,209 @@ rr_sp = resample(
...
 
-The `r ref("mlr_resamplings_spcv_block", text = "spcv_block")` method makes use of rectangular blocks to divide the study area into equally-sized parts.
-These blocks can be visualized by their spatial location and fold ID to get a better understanding how these influenced the final partitions.
+This examples showcases another SpCV method: `r ref("mlr_resamplings_spcv_block", text = "spcv_block")`.
+This method makes use of rectangular blocks to divide the study area into equally-sized parts.
+{mlr3spatiotempcv} has support for visualizing the created blocks and displaying their respective fold ID to get a better understanding how the final folds were composed out of the partitions.
+E.g. the "Fold 1 Repetition 1" plot shows that the test set is composed out of two "blocks" with the ID "1" in this case.
+
+:::{.callout-note}
+The use of `range = 1000L` is arbitrary here and should not be c/p into a real application.
```
Comment (be-marc) [inline book/08-special-spatiotemp.qmd:230]: c/p -> copy-pasted

**Review by be-marc — CHANGES_REQUESTED** [review]

File: book/08-special-spatiotemp.qmd [hunk 3]
```diff
@@ -132,70 +167,209 @@ rr_sp = resample(
 rr_sp$aggregate(measures = msr("classif.ce"))
 ```
 
-Here, the classification tree learner is around 0.05 percentage points worse when using Spatial Cross-Validation (SpCV) compared to Non-Spatial Cross-Validation (NSpCV).
-The magnitude of this difference is variable as it depends on the dataset, the magnitude of STAC and the learner itself.
+Here, the performance of the classification tree learner is around 0.05 percentage points worse when using Spatial Cross-Validation (SpCV) compared to Non-Spatial Cross-Validation (NSpCV).
```
[the remark this answers is not in the document]
Comment (pat-s) [reply book/08-special-spatiotemp.qmd:170]: Good catch!

File: book/08-special-spatiotemp.qmd [hunk 4]
```diff
@@ -17,45 +17,57 @@ Hence, methods specifically designed to account for the special nature of spatio
...
+Two child classes of `r ref("Task")` exist in `r mlr3book::mlr_pkg("mlr3spatiotempcv")` for this purpose:
 
 - `r ref("TaskClassifST")`
 - `r ref("TaskRegrST")`
 
-To create one of these, one can either pass a `sf` object as the "backend" directly:
+To create one of these, you have multiple options:
 
-```r
+1. Use the constructor of the `r ref("Task")` directly via `$new()`
```
[the remark this answers is not in the document]
Comment (pat-s) [reply book/08-special-spatiotemp.qmd:37]: Ok I've removed the contructor option.

**Comment by pat-s** [comment]
@be-marc Please approve and squash merge if ok 👍

**Comment by pat-s** [comment]
Merging since you are on holiday.

**Pull request merged**